examples/run_nlp_trainer.py

```diff
@@ -0,0 +1,350 @@
+
+    s3 = session.resource('s3')
+    bucket = s3.Bucket(bucket_name)
+    ckpt_names = []
+    for obj in bucket.objects.all():
+        if folder_name in obj.key:
+            obj_name = obj.key
+            ckpt_names.append(obj_name)
+
+    return ckpt_names
+
+''' Consolidate and prettify metrics. '''
+def output_metrics() -> None:
+    tasks = GLUE_METRICS[list(GLUE_METRICS.keys())[0]].keys()
+    large_tasks = ['MNLI', 'QNLI', 'QQP', 'SST-2']
+    # init table headers 
+    headers = ['Checkpoint']
+    headers.extend([f"{task.upper()}" for task in tasks])
+    headers.extend(['GLUE-Large', 'GLUE-All'])
+    tb = [headers]
+
+    # fill table 
+    for ckpt in GLUE_METRICS.keys():
+        output_line = [ckpt]
+        glue_all = 0
+        glue_large = 0
+        # Per task score
+        for task in sorted(GLUE_METRICS[ckpt].keys()):
+            output_line.append("{:.2f}".format(GLUE_METRICS[ckpt][task]))
+            glue_all += GLUE_METRICS[ckpt][task]
+            if task in large_tasks:
+                glue_large += GLUE_METRICS[ckpt][task]
+        # GLUE Large and GLUE All
+        output_line.append("{:.2f}".format(glue_large/len(large_tasks)))
+        output_line.append("{:.2f}".format(glue_all/len(tasks)))
+        tb.append(output_line)
+    
+    print(tabulate(tb, headers='firstrow'))
+
+''' Set up CUDA environment and process pool for given finetuning jobs. '''
+def setup_finetuning_jobs(finetune_tasks: List[str], save_ckpts: List[bool], ckpt_load_path: str, ckpt_filename: str,
+                                         save_folder: str, save_locally: bool, bucket: Optional[str] = None, parent_ckpt: Optional[str] = None, load_ignore_keys: Optional[List[str]] = None) -> None:
+    cuda_envs = init_cuda_queue(len(finetune_tasks))
+    num_tasks = len(finetune_tasks)
+    
+    ''' Callback function for metric collection. '''
+    def log_metrics(metric) -> None:
+        if parent_ckpt: 
+            logged_ckpt_name = parent_ckpt
+        else:
+            logged_ckpt_name = ckpt_filename
+
+        if logged_ckpt_name not in GLUE_METRICS.keys():
+                GLUE_METRICS[logged_ckpt_name] = {}
+        task = list(metric.keys())[0]
+        for m in metric[task]:
+            formatted_task = list(metric.keys())[0].split('_')[1] # remove "glue_" prefix
+            if formatted_task not in GLUE_METRICS[logged_ckpt_name].keys():
+                GLUE_METRICS[logged_ckpt_name][formatted_task] = metric[task][m].item()
+            else:
+                GLUE_METRICS[logged_ckpt_name][formatted_task] = sum([GLUE_METRICS[logged_ckpt_name][formatted_task], metric[task][m].item()]) / 2  # automatically average scores 
+        
+    if parent_ckpt: 
+        wandb_group_name = parent_ckpt
+    else:
+        wandb_group_name = ckpt_load_path
+
+    # finetuning from pretrained checkpoint
+    print(f"FINETUNING ON {ckpt_filename}!")
+    pool = mp.Pool(num_tasks, initializer=init_cuda_env, initargs=(cuda_envs,))
```

Comment:
Fancy use of multiprocessing :)

```suggestion
    pool = mp.Pool(torch.cuda.device_count(), initializer=init_cuda_env, initargs=(cuda_envs,))
```

I presume that each mp worker will use the same device id, even if it is reused?